OasisDEX/oasis-borrow
# Prepare manage borrow vault pipeline for insti vaults

Builds off Damian's work and prepares manage borrow pipeline to support charter (institutional) vaults, and integrates into the view created by Ferni.

There's no user-facing change at present.  Technical notes:

- new `InstiVault$` that wraps standard Vault and reads from additional smart contract.
- `createManageVault$` is used for both `manageVault$` and `manageInstiVault$`.  `createManageVault$` receives initial view state and `apply` function as a dependency, and there is one instance of `BorrowManageVaultViewStateProviderInterface` for `manageVault$` and one for `manageInstiVault$`.
- `ManageVaultState` renamed to `ManageStandardBorrowVaultState`.
- provides a place in AppContext to hardcode vault IDs which we can use if we can't determine vault type (`Borrow | Insti`) from the smart contracts (perhaps DB is better to avoid this logic?)
- `features/borrow/manage/tests/manageVault.test.ts` looks a bit intense, but it's just repeating the same tests for `manageVault$` and `manageInstiVault$` so that the same user interactions work for both, and adding some other insti vault specific things.

Testing:

Can't currently test managing insti vaults as we don't have smart contracts deployed, but we can create a borrow vault and then flag it as an insti vault:

1. create a standard borrow vault.
2. mark it as an insti vault in `HARDCODED_VAULT_TYPES` in AppContext
3. Set the `withdrawPaybackDepositGenerateLogicFactory` to `StandardDssProxyActionsContractWrapper` on line 513-515.
4. When creating `instiVault$` in AppContext, use the dummy pipes that are commented out rather than `charterNib$, charterPeace$, charterUline$`
5. Refresh the page.
6. You should now see the insti vault UI and be able to interact with the vault in the normal way.

- Also need regression testing on manage borrow vault before deploy.

Next steps (future PR):
- hydrate InstiVault termEnd, fixedFee, and nextFixedFee when we get the new smart contracts that fullfill these.
- determine institutional vault types either with a smart contract call or by populating HARDCODED_VAULT_TYPES
